src/general_manager/measurement/measurementField.py

```
@@ -1,168 +1,275 @@
+            if value.quantity.dimensionality != self.base_dimension:
+                raise ValidationError(
+                    {self.name: [f"Unit must be compatible with '{self.base_unit}'."]}
+                )
+
+        try:
+            base_mag = value.quantity.to(self.base_unit).magnitude
+        except pint.errors.DimensionalityError:
             raise ValidationError(
-                {self.name: ["Value must be a Measurement instance or None."]}
+                {self.name: [f"Unit must be compatible with '{self.base_unit}'."]}
             )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)